src/plugins/tooltip/directives/directive.ts

```diff
@@ -0,0 +1,173 @@
+import { DirectiveBinding } from 'vue';
+import { createPopper, Placement, Instance } from '@popperjs/core';
+import '../styles/styles.scss';
+
+import {
+  CLASS_TOOLTIP,
+  CLASS_ANCHOR,
+  CLASS_TOOLTIP_WRAPPER,
+  CLASS_TOOLTIP_WRAPPER_ACTIVE,
+  CLASS_TOOLTIP_WRAPPER_VISIBLE,
+  DEBOUNCE_CLOSE,
+  DEFAULT_POPPER_OPTIONS,
+} from '@/plugins/tooltip/shared';
+
+type TTooltipEvent = {
+  [key: string]: EventListener;
+}
+
+interface ITooltipElement extends HTMLElement {
+  [key: symbol]: {
```

Comment:
question: would it be possible to use the `storageKey` as a key directly? (Unfortunately, this would break the convention to always have TS on top, but it would be more clear.